src/main/java/codechicken/nei/recipe/RecipeCatalysts.java

```diff
@@ -12,39 +12,49 @@
 import java.util.Map;
 
 public class RecipeCatalysts {
-    private static final Map<Class<? extends IRecipeHandler>, List<ItemStack>> recipeCatalystMap = new HashMap<>();
-    private static Map<Class<? extends IRecipeHandler>, List<PositionedStack>> positionedRecipeCatalystMap = new HashMap<>();
+    private static final Map<String, List<ItemStack>> recipeCatalystMap = new HashMap<>();
+    private static Map<String, List<PositionedStack>> positionedRecipeCatalystMap = new HashMap<>();
     private static int heightCache;
 
-    public static void addRecipeCatalyst(List<ItemStack> stacks, Class<? extends IRecipeHandler> handler) {
-        if (recipeCatalystMap.containsKey(handler)) {
-            recipeCatalystMap.get(handler).addAll(stacks);
+    public static void addRecipeCatalyst(List<ItemStack> stacks, String handlerID) {
+        if (handlerID == null) return;
+        if (recipeCatalystMap.containsKey(handlerID)) {
+            recipeCatalystMap.get(handlerID).addAll(stacks);
         } else {
-            recipeCatalystMap.put(handler, new ArrayList<>(stacks));
+            // use ArrayList initializer to prevent UOE
+            recipeCatalystMap.put(handlerID, new ArrayList<>(stacks));
         }
     }
 
-    public static Map<Class<? extends IRecipeHandler>, List<PositionedStack>> getPositionedRecipeCatalystMap() {
+    public static Map<String, List<PositionedStack>> getPositionedRecipeCatalystMap() {
```

Comment:
This seems a change to a public map, should it be wrapped somehow?

Reply:
Since this change was made in #195, I assume nobody has implemented this in the last 11 hours.

Reply:
hehehehe you'd be surprised but yeah

Reply:
okay I'll do it

Reply:
nah I'm being silly, 11hrs on your own code is fine 😄

Reply:
IDEA says `'getPositionedRecipeCatalystMap()' is already defined in 'codechicken.nei.recipe.RecipeCatalysts`, though I changed signature properly.

Reply:
you may have to refresh IDEA after changing things; it can cache lots and miss some changes.

Reply:
And compile fails too :(